Model/Category/Repository.php

```diff
@@ -0,0 +1,186 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2020 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Model\Category;
+
+use Magento\Catalog\Api\Data\CategoryInterface;
+use Magento\Catalog\Model\Category;
+use Magento\Catalog\Model\CategoryRepository;
+use Magento\Catalog\Model\ResourceModel\Category\Collection as CategoryCollection;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Store\Model\Store;
+use Magento\Catalog\Model\ResourceModel\Category\CollectionFactory as CategoryCollectionFactory;
+use Nosto\Tagging\Exception\ParentCategoryDisabledException;
+
+/**
+ * Repository wrapper class for fetching categories
+ */
+class Repository
+{
+    private array $parentCategoryIdCache = [];
+
+    /** @var SearchCriteriaBuilder $searchCriteriaBuilder */
+    private SearchCriteriaBuilder $searchCriteriaBuilder;
+
+    /** @var CategoryRepository $categoryRepository */
+    private CategoryRepository $categoryRepository;
+
+    /** @var CategoryCollectionFactory $categoryCollectionFactory */
+    private CategoryCollectionFactory $categoryCollectionFactory;
+
+    /**
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     */
+    public function __construct(
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        CategoryRepository $categoryRepository,
+        CategoryCollectionFactory $categoryCollectionFactory
+    ) {
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->categoryRepository = $categoryRepository;
+        $this->categoryCollectionFactory = $categoryCollectionFactory;
+    }
+
+    /**
+     * Gets categories by category ids
+     *
+     * @param array $ids
+     */
+    public function getByIds(array $ids)
+    {
+        //@TODO implement
+    }
+
+    /**
+     * @param Store $store
+     * @param array $categoryIds
+     *
+     * @return CategoryCollection
+     * @throws LocalizedException
+     */
+    public function getCategoryCollectionQuery(Store $store, array $categoryIds = [])
+    {
+        $categories = $this->categoryCollectionFactory->create()
+            ->distinct(true)
+            ->addNameToResult()
+            ->setStoreId($store->getId())
+            ->addUrlRewriteToResult()
+            ->addAttributeToFilter('level', ['gt' => 1]) // @TODO: Check if zero level categories are needed
```

Comment:
That's OK, level 1 is fine, you may remove the comment
```suggestion
            ->addAttributeToFilter('level', ['gt' => 1])
```